### ISSUE TITLE
Change serviceMonitors of scheduler and controller-manager to podMonitor

### ISSUE THREAD

**Issue by yeya24** [coreos/kube-prometheus#169]
Since prometheus-operator already supports podMonitor, is it better to migrate serviceMonitor to podMonitor? 

**What did you do?**
`kubectl apply -f /manifests`
Open prometheus-ui and find cannot get target for controller-manager and scheduler.
![深度截图_选择区域_20190725130309](https://user-images.githubusercontent.com/25150124/61847265-948fc900-aedc-11e9-9f15-b59f84b53bf2.png)

So in order to get the metrics I have to create controller and scheduler Service. If it is podMonitor then I guess I don't need to expose these two Services.

**Did you expect to see some different?**

**Environment**

* Prometheus Operator version:

    `Insert image tag or Git SHA here`
    

* Kubernetes version information:

    `1.15.0`

* Kubernetes cluster kind:

    `kubeadm init`

* Manifests:

 master branch `manifests/`

* Prometheus Operator Logs:

```
Insert Prometheus Operator logs relevant to the issue here
```

* Prometheus Logs:

```
Insert Prometheus logs relevant to the issue here
```

**Anything else we need to know?**:

**Comment by metalmatze** [comment]
This might be an issue with labels not matching too.
However, if we can directly scrape these, then I'm all for using PodMonitors here. :)

**Comment by yeya24** [comment]
The service monitor definition is in https://github.com/coreos/kube-prometheus/blob/master/jsonnet/kube-prometheus/prometheus/prometheus.libsonnet. If we want to change service monitor to pod monitor, then all examples https://github.com/coreos/kube-prometheus/tree/master/jsonnet/kube-prometheus will be affected.

This change is too breaking MO. WDYT?

**Comment by brancz** [comment]
I don’t think this is breaking, as long as we keep the same selectors for pods in the end. The problem that I can see is that PodMonitors don’t support TLS and authentication pieces today which would be necessary for controller manager and scheduler metrics scraping.
